sdk/communication/AzureCommunicationChat/Source/ChatThreadClient.swift

```diff
@@ -0,0 +1,455 @@
+                withOptions: options
+            ) { result, httpResponse in
+                switch result {
+                case .success:
+                    completionHandler(.success(()), httpResponse)
+
+                case let .failure(error):
+                    completionHandler(.failure(error), httpResponse)
+                }
+            }
+    }
+
+    /// Lists read receipts for the ChatThread.
+    /// - Parameters:
+    ///   - options: List chat read receipts options.
+    ///   - completionHandler: A completion handler that receives the list of read receipts on success.
+    public func listReadReceipts(
+        withOptions options: ChatThreadOperation.ListChatReadReceiptsOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<PagedCollection<ReadReceipt>>
+    ) {
+        service.listChatReadReceipts(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case .success:
+                // TODO: https://github.com/Azure/azure-sdk-for-ios/issues/644
+                // Construct a new PagedCollection of type ReadReceipt
+                do {
+                    let readReceipts = try self.createPagedCollection(
+                        from: httpResponse?.data,
+                        withRequest: httpResponse?.httpRequest,
+                        of: ReadReceipt.self
+                    )
+
+                    completionHandler(.success(readReceipts), httpResponse)
+                } catch {
+                    let azureError = AzureError.client(error.localizedDescription, error)
+                    completionHandler(.failure(azureError), httpResponse)
+                }
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a typing notification.
+    /// - Parameters:
+    ///    - options: Send typing notification options
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func sendTypingNotification(
+        withOptions options: ChatThreadOperation.SendTypingNotificationOptions? = nil,
+        completionHandler: @escaping HTTPResultHandler<Void>
+    ) {
+        service.sendTypingNotification(chatThreadId: threadId, withOptions: options) { result, httpResponse in
+            switch result {
+            case .success:
+                completionHandler(.success(()), httpResponse)
+
+            case let .failure(error):
+                completionHandler(.failure(error), httpResponse)
+            }
+        }
+    }
+
+    /// Sends a message to a ChatThread.
+    /// - Parameters:
+    ///    - message : Request that contains the message properties.
+    ///    - options: A list of options for the operation.
+    ///    - completionHandler: A completion handler that receives a status code on success.
+    public func send(
+        message: SendChatMessageRequest,
```

Comment:
There's an opportunity to streamline the customer experience by unwrapping Send/UpdateChatMessageRequest (potentially). I know we discussed it before, but didn't settle on the final form. What does this look like in usage?

Reply:
Yes - I streamlined on some other calls for example `chatThreadClient.update(topic: "new topic")` so we avoid having to pass an UpdateChatThreadRequest 

The message request looks like this, only content is required:
```
       let messageRequest = SendChatMessageRequest(
            content: "Hello",
            senderDisplayName: "Kim",
            type: .text
        )
        
        chatThreadClient.send(message: messageRequest) { ...
```

Reply:
Okay, so long as you are happy with it. You can always tweak it based on feedback since it's a preview.